src/pages/api/auth/[...nextauth].ts

```diff
@@ -1,15 +1,34 @@
+import createFetch from '@vercel/fetch'
 import NextAuth from 'next-auth'
 import CloudIdpProvider from 'lib/auth/cloud-idp-provider'
 import refreshTokenSet from 'lib/auth/refresh-token-set'
 import isJwtExpired from 'lib/auth/is-jwt-expired'
 import { AuthErrors } from 'types/auth'
 
+const fetch = createFetch()
+
 const isDev = process.env.NODE_ENV === 'development'
 
 export default NextAuth({
 	session: {
 		maxAge: 2592000, // 30 days
 	},
+	// https://next-auth.js.org/configuration/options#events
+	events: {
+		/**
+		 * NOTE: NextAuth log out of auth providers, so we have to handle doing that
+		 * ourselves in this signOut event.
+		 * https://github.com/nextauthjs/next-auth/discussions/3938
+		 *
+		 * cloud-idp and Ory Hydra docs:
+		 * https://hashicorp.atlassian.net/wiki/spaces/CLOUD/pages/2321449597/cloud-idp+and+Ory+Hydra
```

Comment:
As this repo is open source, it might be best if we don't link to internal references like this. Maybe we can explain that this is the `end_session_endpoint` from `.well-known` file?

Reply:
Updated here: https://github.com/hashicorp/dev-portal/pull/1301/commits/aadc63ce12859d807812f58876295ea0fe200dc4